src/custom/state/lists/updater.tsx

```diff
@@ -0,0 +1,104 @@
+        // access using chainId
+        const newListOfLists = DEFAULT_LIST_OF_LISTS_BY_NETWORK[chainId]
+        const newActiveListUrls = DEFAULT_ACTIVE_LIST_URLS_BY_NETWORK[chainId]
+
+        // Make a copy of the previous storage list
+        const tokenStateCopy = Object.assign({}, oldStorageList)
+
+        // we need to compare the lists and merge/cleanup
+        // lets start with the last initialised list of lists
+        if (oldStorageList.lastInitializedDefaultListOfLists) {
+          // Lets create a Set of the last initialised list of lists saved, to compare
+          const oldStorageLastInitializedSet: Set<string> = new Set(tokenStateCopy.lastInitializedDefaultListOfLists)
+          // and a set of the (potentially) new ones from 'custom/constants/lists'
+          const newListofListsSet = new Set(newListOfLists)
+
+          // lets loop over the 'constants/lists' listsOfLists and check the
+          // oldStorageLastInitializedSet vs the storage list one
+          newListofListsSet.forEach(newListUrl => {
+            // if storage token list doesn't have a token list URL
+            // that the new default does, we add it
+            if (!oldStorageLastInitializedSet.has(newListUrl)) {
+              tokenStateCopy.byUrl[newListUrl] = NEW_LIST_STATE
+            }
+          })
+
+          // check the previous lists "lastInitialisedTokens" against the incoming list
+          oldStorageLastInitializedSet.forEach((oldListUrl: string) => {
+            // remove any token list urls that the default list doesn't have
+            if (!newListofListsSet.has(oldListUrl)) {
+              delete tokenStateCopy.byUrl[oldListUrl]
+            }
+          })
+
+          // if no active lists, activate defaults
+          if (!tokenStateCopy.activeListUrls) {
+            tokenStateCopy.activeListUrls = newActiveListUrls
+
+            // for each list on default list, initialize if needed
+            newActiveListUrls.map((listUrl: string) => {
+              if (!tokenStateCopy.byUrl[listUrl]) {
+                tokenStateCopy.byUrl[listUrl] = NEW_LIST_STATE
+              }
+              return true
+            })
+          }
+        }
+
+        // set our state's last initialised to the new list
+        tokenStateCopy.lastInitializedDefaultListOfLists = newListOfLists
+
+        refreshedTokenState = tokenStateCopy
+      } else {
+        // clean slate, build defaults
+        refreshedTokenState = buildListState({
+          defaultActiveListUrls: DEFAULT_ACTIVE_LIST_URLS_BY_NETWORK[chainId],
+          defaultListofLists: DEFAULT_LIST_OF_LISTS_BY_NETWORK[chainId],
+          newListState: NEW_LIST_STATE
+        })
+      }
+
+      batchedUpdates(() => {
+        // save chain aware tokens to localStorage and redux
+        dispatch(initialiseTokenLists(refreshedTokenState))
+        localStorage.setItem(newChainKey, JSON.stringify(refreshedTokenState))
```

Comment:
Is the responsibility of the updater to persist in the local storage?
Why not using the middleware as Uniswap does for all the others?

I feel this is otherwise hidden in this logic.
The updater's responsibility is to dispatch actions when things happens. In this case, a network change.

Reply:
the middleware cannot update it individually with the appended network key e.g `redux_simple_storage_lists_4` as it uses the key name given which maps to the redux state. as I am not changing the redux state shape at all, we cannot use that middleware as it stands now, it just won't work. It will however, save the non-network specific list currently loaded, e.g `redux_simple_storage_lists`

Reply:
but in the end I chose to not over-saturate saving things to storage across several places as race conditions etc. This at least keeps our lists saving logic to storage in one place